lib/plugins/aws/package/compile/events/kafka.js

```diff
@@ -167,6 +177,14 @@ class AwsCompileKafkaEvents {
                 type = 'SASL_SCRAM_512_AUTH';
                 needsSecretsManagerPermissions = true;
                 break;
+              case 'clientCertificateTLSAuth':
+                type = 'CLIENT_CERTIFICATE_TLS_AUTH';
+                needsSecretsManagerPermissions = true;
+                break;
+              case 'serverRootCaCertificate':
```

Comment:
I have a question about this property - does it make sense/is it effective if `clientCertificateTlsAuth` is not used? My understanding was that you can provide CA Cert if the client certs are signed by this private CA, are there any other use cases that I'm missing?

Reply:
Ahh good point. I'm not sure what happens if you provide a ca cert with no client cert. I can do some experiments and find out. If you are correct (I expect you are) and the cloudformation fails, do you think we should apply validation of these settings upfront?

Reply:
I think it would be great to verify what happens in that situation. To be honest, I'm not sure if it's going to fail, I believe it might accept it, but it will be simply innefective. If both properties are required, then we should validate these settings up front instead of relying on CloudFormation to error out.

Reply:
OK, I did some investigation here. With the following Cloudformation template:

```yaml
Resources:
  MyEventSourceMapping:
    Type: AWS::Lambda::EventSourceMapping
    Properties:
      FunctionName: arn:aws:lambda:eu-west-1:xxxxxxxx:function:test
      StartingPosition: 'TRIM_HORIZON'
      Topics: 
        - "MyTopic"
      SelfManagedEventSource: 
        Endpoints: 
          KafkaBootstrapServers: 
            - abc.xyz.com:9092
      SourceAccessConfigurations: 
        - Type: "SERVER_ROOT_CA_CERTIFICATE" 
          URI: "arn:aws:secretsmanager:eu-west-1:xxxxxxxx:secret:test-ca-cert-xxxxx"
```

The stack will not deploy, with the error `Resource handler returned message: "Invalid request provided: Unsupported source access configurations type for given event source mapping type. (Service: Lambda, Status Code: 400, Request ID: b4526f54-c294-4e45-abd9-fa171eca6b12, Extended Request ID: null)" (RequestToken: f214e871-ad1a-4d3a-f5b0-655827d8ad09, HandlerErrorCode: InvalidRequest)`

Modifying the template so that a `CLIENT_CERTIFICATE_TLS_AUTH` is defined will allow the stack to deploy

Reply:
So I think that Serverless should mirror this validation. Also, stack will not deploy if `SelfManagedEventSource`/`Topics` is defined, but no `SourceAccessConfigurations` are defined

Reply:
I've added in validations via schema to this effect now 🙇

Reply:
Thanks a lot for research, we definitely want to validate that in such case 💯

Reply:
no probs. looks like we will change approach from schema validation though